cartography/stats.py

```diff
@@ -53,3 +53,24 @@ def timer(self, stat: str, rate: float = 1.0):
         if self.is_enabled():
             return self._client.timer(stat, rate)
         return None
+
+
+# Global _stats_client
+# Will be set when cartography.config.statsd_enabled is True
+_stats_client: StatsClient = None
+
+
+def set_stats_client(stats_client):
+    """
+    This is used to set the module level stats client configured to talk with a statsd host
+    """
+    global _stats_client
```

Comment:
I'm scared of globals. Is this absolutely necessary? Can we accomplish the scenario you have in mind without this?

Reply:
Isn't cartography.util.stats_client also global? The _stats_client is exactly what cartography.util.stats_client used to be (statsd client). The set_stats_client is just a setter to initialize the  _stats_client (will happen as part of the sync run)

Reply:
Any reason we can't just create a stats client object in the cli/main, and pass it to sync? Or at least have a singleton stats client class  that we provide a handle to a la `cartography.util.stats_client`?

For example, we should only need to do this once:
```
        cartography.util.stats_client = StatsClient(
            host=config.statsd_host,
            port=config.statsd_port
        )
```
Then after we can call that with get_stats_client(prefix).

Reply:
Payal and I talked about this a bit yesterday. Internally we have other stats code that follows this same pattern of using a global variable for the stats client, so I'm fine with doing the same thing here.

> Any reason we can't just create a stats client object in the cli/main, and pass it to sync?

Do you mean adding an object to the config object, or adding another parameter to a sync_*/start_* function? The latter is a much, much larger refactor.

Reply:
I was thinking of attaching to the config object. I'm totally fine with the current approach if you are though. Better we get things working now in a minimal way, and then we figure out  the bigger problem!